design/sketch.webidl

```diff
@@ -169,12 +169,12 @@ interface GPUTextureUsage {
 
 dictionary GPUTextureDescriptor {
     GPUExtent3D size;
-    u32 arrayLayerCount;
-    u32 mipLevelCount;
-    u32 sampleCount;
-    GPUTextureDimension dimension;
-    GPUTextureFormat format;
-    GPUTextureUsageFlags usage;
+    u32 arrayLayerCount = 1;
+    u32 mipLevelCount = 1;
+    u32 sampleCount = 1;
+    GPUTextureDimension dimension = "2d";
+    GPUTextureFormat format = "rgba8unorm";
+    GPUTextureUsageFlags usage = GPUTextureUsage.TRANSFER_SRC | GPUTextureUsage.TRANSFER_DST | GPUTextureUsage.SAMPLED | GPUTextureUsage.Storage | GPUTextureUsage.OUTPUT_ATTACHMENT;
```

Comment:
Having a default for `usage` doesn't really make sense to me - when would an app ever use the default list? Especially problematic if the default might cause deoptimization (e.g. by having STORAGE and OUTPUT_ATTACHMENT which iirc prevent compression and possibly other things).

Also, `Storage` is capitalized incorrectly. (btw, is this valid default value syntax?)

Reply:
Of course there's a chance for deoptimization; if there wasn't a chance for it, we wouldn't have the `usage` field in the first place.

The purpose of defaults is to make developers' lives easier. A developer who is just starting out will use the defaults to make their app directly, whereas a more experienced developer will to the necessary resource tracking to optimize these usage flags. Allowing for the most flexible behavior, even at the expense of some deoptimization, is the best default we can pick.

Reply:
We should probably add GPUTextureUsage.ALL and use that instead.

Reply:
Now if a developer accidentally forgets to set the field, they will get a slow app instead of an error. That doesn't sound right to me. If we want to give beginners an easy way to start, we can give them higher-level frameworks on top that get the details right, or tell them to opt-in to the slow thing while they get their app and running, and hope that they fix it later. Don't set beginners up for failure.

Reply:
I agree with that. I also think that forcing devs to look at the list of usages and pick some is not overly burdensome, even to beginners.

Reply:
TRASNFER_SRC and/or SAMPLED might be the highest-likelyhood default.